src/Microsoft.DotNet.Build.Tasks/DownloadFileFromUrl.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class DownloadFileFromUrl : BuildTask
+    {
+        
+        /// <summary>
+        /// The url to download the file from.
+        /// </summary>
+        [Required]
+        public string DownloadSource { get; set;}
+        
+        /// <summary>
+        /// The directory where the file will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationDirectory { get; set; }
```

Comment:
If I have to specify both anyway, why not just have Destination that takes a path? Taht way if I have a path, I don't have to split it to pass it in.

Reply:
Makes sense. I did it to make sure that the directory existed before creating the file, but I can achieve that by getting the parent directory from the full path, and create it before creating the file. (This is because File.Create throws if part of the path doesn't exist)